activestorage/test/models/blob_test.rb

```diff
@@ -261,12 +261,12 @@ class ActiveStorage::BlobTest < ActiveSupport::TestCase
   end
 
   private
-    def expected_url_for(blob, disposition: :attachment, filename: nil, content_type: nil)
+    def expected_url_for(blob, disposition: :attachment, filename: nil, content_type: nil, service_name: :local)
       filename ||= blob.filename
       content_type ||= blob.content_type
 
       query = { disposition: ActionDispatch::Http::ContentDisposition.format(disposition: disposition, filename: filename.sanitized), content_type: content_type }
-      key_params = { key: blob.key }.merge(query)
+      key_params = { key: blob.key }.merge(query).merge(service_name: service_name)
```

Comment:
This weirdness is due to the following behavior of `ActiveSupport::MessageVerifier`:
```ruby
(byebug) ActiveStorage.verifier.generate({a: 'a', b: 'b'})
"BAh7BzoGYUkiBmEGOgZFVDoGYkkiBmIGOwZU--e93361b08967d5f39dfcd54913bf9b5639daafad"
(byebug) ActiveStorage.verifier.generate({b: 'b', a: 'a'})
"BAh7BzoGYkkiBmIGOgZFVDoGYUkiBmEGOwZU--8c7a82678420675b0883662e293e2b3f9d8e1c70"
```
So the generated key seems to depend on the order of the keys of the hash. Not sure if this is intended behavior.

Reply:
The hash is serialized so unless we sort the hash before we dump it the key order is maintained and the generated string will be different. I think this is fine considering its only a problem during comparison of encoded values.